crates/oxide/src/glob.rs

```diff
@@ -136,6 +139,14 @@ fn get_fast_patterns(base_path: &Path, patterns: &Vec<String>) -> Vec<(PathBuf,
     optimized_patterns
 }
 
+pub fn path_matches_globs(path: &Path, globs: &[GlobEntry]) -> bool {
```

Comment:
I feel like this is going to be useful for a bunch of other things as well (e.g. the `node_module` should warn unless the glob it matches contains `node_modules` idea we chatted about last week).